plugins/network-elements/nuage-vsp/src/com/cloud/network/resource/NuageVspResource.java

```diff
@@ -441,6 +456,44 @@ private Answer executeRequest(SyncVspCommand cmd) {
         }
     }
 
+    private Answer executeRequest(SyncNuageVspCmsIdCommand cmd) {
+        try {
+            isNuageVspManagerLoaded();
+            if (cmd.getSyncType() == SyncType.AUDIT || cmd.getSyncType() == SyncType.AUDIT_ONLY) {
+                Pair<Boolean, String> answer = _nuageVspManagerClient.auditNuageVspCmsId(cmd.getNuageVspCmsId(), cmd.getSyncType() == SyncType.AUDIT_ONLY);
+                return new SyncNuageVspCmsIdAnswer(answer.getLeft(), answer.getRight(), cmd.getSyncType());
+            } else if (cmd.getSyncType() == SyncType.REGISTER) {
+                String registeredNuageVspCmsId = _nuageVspManagerClient.registerNuageVspCmsId();
+                return new SyncNuageVspCmsIdAnswer(StringUtils.isNotBlank(registeredNuageVspCmsId), registeredNuageVspCmsId, cmd.getSyncType());
+            } else {
+                boolean success = _nuageVspManagerClient.unregisterNuageVspCmsId(cmd.getNuageVspCmsId());
+                return new SyncNuageVspCmsIdAnswer(success, cmd.getNuageVspCmsId(), cmd.getSyncType());
+            }
+        } catch (Exception e) {
```

Comment:
Why are all unchecked and checked exceptions being caught?  It seems like we should only be catching the expected checked exceptions, and allow unchecked exceptions simply bubble out.